src/parquet/column/serialized-page.cc

```diff
@@ -45,23 +47,43 @@ SerializedPageReader::SerializedPageReader(std::unique_ptr<InputStream> stream,
   }
 }
 
-// TODO(wesm): this may differ from file to file
-static constexpr int DATA_PAGE_SIZE = 64 * 1024;
 
 std::shared_ptr<Page> SerializedPageReader::NextPage() {
   // Loop here because there may be unhandled page types that we skip until
   // finding a page that we do know what to do with
   while (true) {
+    int bytes_available = 0;
     int bytes_read = 0;
-    const uint8_t* buffer = stream_->Peek(DATA_PAGE_SIZE, &bytes_read);
-    if (bytes_read == 0) {
-      return std::shared_ptr<Page>(nullptr);
-    }
-
-    // This gets used, then set by DeserializeThriftMsg
-    uint32_t header_size = bytes_read;
-    DeserializeThriftMsg(buffer, &header_size, &current_page_header_);
+    uint32_t header_size = 0;
+    const uint8_t* buffer;
+    int allowed_page_size = page_header_size_;
```

Comment:
set the first allowed_page_size to the initial expected page header size constant here